### DISCUSSION TITLE
Aspire Azure Entra External ID integration

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#6206] [Ideas]
Just wondering if there was anything in the works for an Aspire Azure Entra External ID integration?

Would be great in AppHost if you could define an external ID directory, connect it with your web apps and have aspire/azd deploy/configure it all.

**Comment by davidfowl** [comment]
Can't wait to get there, but we are not there yet. Now that it is possible to perform entra provisioning with bicep more of this will be possible.

**Reply by shawnwildermuth** [reply]
Any news on this?

**Comment by ahelland** [comment]
@mip1983 I would also think the Graph support in Bicep is the way to go. I have not tested scripting up External ID yet (I probably should), but with regular Entra ID it's possible to wire into Aspire:
https://github.com/ahelland/azd-customizations/blob/main/03_End-to-End/BFF_Aspire/infra/Graph/app-registration.bicep

**Reply by mip1983** [reply]
> @mip1983 I would also think the Graph support in Bicep is the way to go. I have not tested scripting up External ID yet (I probably should), but with regular Entra ID it's possible to wire into Aspire: https://github.com/ahelland/azd-customizations/blob/main/03_End-to-End/BFF_Aspire/infra/Graph/app-registration.bicep

Thanks for sharing that, that'll make a good jumping off point to try codify some of the Entra setup I've got, particularly as I'm not super hot on the bicep/powershell side of things.
